src/pycountry/__init__.py

```diff
@@ -186,12 +204,12 @@ def parent(self):
         return subdivisions.get(code=self.parent_code)
 
 
-class Subdivisions(pycountry.db.Database):
+class Subdivisions(pycountry.db.Database[SubdivisionHierarchy]):
     # Note: subdivisions can be hierarchical to other subdivisions. The
     # parent_code attribute is related to other subdivisions, *not*
     # the country!
 
-    data_class = SubdivisionHierarchy
+    factory = SubdivisionHierarchy
     no_index = ["name", "parent_code", "parent", "type"]
     root_key = "3166-2"
 
```

Comment:
This comment is actually in reference to line 265. Type hinting for `Subdivisions.search_fuzzy()` needs updating per @terencehonles's comments.

Adding this as a change request before merging for tracking purposes.

Reply:
I've updated this and made some changes so there should be a complete review on this since I made modifications based on errors mypy was pointing out.